modules/4337/contracts/Safe4337Module.sol

```diff
@@ -211,6 +211,41 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
         operationHash = keccak256(operationData);
     }
 
+    /**
+     * @dev Checks if the signatures length is correct and does not contain additional bytes. The function does not check the integrity of the signature encoding.
+     * @param signatures signatures data
+     * @param threshold Indicates the number of iterations to perform in the loop.
+     * @return result True if length check passes, false otherwise.
+     */
+    function _checkSignatureLength(bytes calldata signatures, uint256 threshold) internal pure returns (bool result) {
+        uint256 offset = threshold * 0x41;
+
+        if (signatures.length < offset) return false;
+
+        result = true;
+        for (uint256 i = 0; i < threshold; i++) {
+            /* solhint-disable no-inline-assembly */
+            /// @solidity memory-safe-assembly
+            assembly {
+                let signaturePos := mul(0x41, i)
+                let s := calldataload(add(signatures.offset, add(signaturePos, 0x20)))
+                let v := byte(0, calldataload(add(signatures.offset, add(signaturePos, 0x40))))
+
+                if iszero(v) {
```

Comment:
I'd add a comment that v = 0 means a smart contract signature in Safe Signature Encoding

Reply:
https://github.com/safe-global/safe-modules/pull/453/commits/af3b7088120302b3b14a9ef74de4c0605a0192f7